src/data/list.js

```diff
@@ -0,0 +1,176 @@
+const mongoCollections = require('../config/mongoCollections');
+const boardData = require('./boards');
+const boards = mongoCollections.boards;
+const {ObjectId} = require('mongodb');
+const error_handler = require('../errors/error-handler'); 
+
+
+module.exports = {
+    /**
+    * Gets all lists from a given board id.
+    * @param {String} id The board's id.
+    * @returns A list of list(column) objects.
+    */
+    getAllListsInBoard: async (id) => {
+        if(!id || !error_handler.checkObjectId(id)) 
+            throw new Error("id is not valid.");
+
+        const board = await boardData.readById(id);
+        let returnList = [];
+        for(let x=0; x<board.lists.length; x++) {
+            let lists = board.lists[x];
+            lists._id = lists._id.toString();
+            returnList.push(lists);
+        }
+
+        return returnList;
+    },
+
+    /**
+    * Gets a list with a given id.
+    * @param {String} listId The lists's id.
+    * @param {String} boardId The board's id.
+    * @returns A list object, if list does not exist, throws error.
+    */
+    getListById: async (listId, boardId) => {
+        if(!listId || !error_handler.checkObjectId(listId)) 
+            throw new Error("listId is not valid.");
+
+        if(!boardId || !error_handler.checkObjectId(boardId)) 
+            throw new Error("boardId is not valid.");
+        
+        const board = await boardData.readById(boardId);
```

Comment:
It's better to not make them dependent on each other because they are modules. So, doing your own calls to the db rather than calling board data. Dk if Hill has a problem with this approach because it goes againt the idea of node to an extent. Look to how you did reviews in lab 6.

Reply:
Rocco agrees

Reply:
This obviusly goes for all of these

Reply:
I thought I copied this approach from him but I might be wrong, its easy enough to change so its fine